plone/namedfile/file.py

```diff
@@ -266,14 +273,24 @@ class NamedImage(NamedFile):
     filename = FieldProperty(INamedFile['filename'])
 
     def __init__(self, data='', contentType='', filename=None):
-        self.contentType, self._width, self._height = getImageInfo(data)
-        self.data = data
         self.filename = filename
+        self._setData(data)
 
         # Allow override of the image sniffer
         if contentType:
             self.contentType = contentType
 
+        if self.contentType in ['image/jpeg', 'image/tiff']:
```

Comment:
This is not valid. Since explicitly setting `self.contentType` was removed some lines above (and `_setData` will only set it if is defined), it can happen that `self.contentType` is undefined. That's where all the test failures are coming from.
I think line 280 should therefore be:

    if contentType or not getattr(self, 'contentType', None):